tracer/src/Datadog.Trace/Debugger/LiveDebuggerFactory.cs

```diff
@@ -49,7 +49,7 @@ public static LiveDebugger Create(IDiscoveryService discoveryService, IRcmSubscr
             () => new MinimalAgentHeaderHelper(),
             uri => uri);
 
-        var batchApi = AgentBatchUploadApi.Create(apiFactory, discoveryService);
+        var batchApi = AgentBatchUploadApi.Create(apiFactory, discoveryService, TracerManager.Instance.GitMetadataTagsProvider);
```

Comment:
nit - should we pass this in as a parameter as well? That would have the upside of making the order in which things need to be initialized a little bit more explicit)

Reply:
I think this is what Omer is saying, but I support passing in the GitMetadataTagsProvider to the `LiveDebuggerFactory.Create` method to limit the amount of static dependencies this method needs